src/app.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useRef, useState } from 'react';
+import List from './list';
+import Form from './form';
+import Button from './button';
+import Input from './input';
+
+function App() {
+  const [toDos, setToDos] = useState([]);
+  const toDoInputs = useRef('');
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    const currentInput = toDoInputs.current.value;
+    const resetInputValue = () => {
+      document.getElementsByClassName('todo-input')[0].value = '';
```

Comment:
React는 컴포넌트 상태 변화가 감지되면 VDOM에 반영하고 비동기적으로 DOM을 조작하기 때문에,
현재 어떤 상태인지 모르는 DOM을 이용하는 DOM API보다 라이프사이클 이후에 DOM 객체가 current에 반영되는
ref를 권장하는 게 아닌가 생각합니다. 혹시 참고할만한 레퍼런스가 있을까요?!

Reply:
> It’s fair to say if you want to send a value on submit, this approach will work, but the issue here is that because we know refs provide an escape hatch of the view model React offers, we can too easily go sniffing into DOM element values or properties that we can access through React’s interface.
[A complete guide to React refs](https://blog.logrocket.com/complete-guide-react-refs/)

정답은 아니지만 가장 공감이 많이가는 부분은 밑에 스택 오버플로우의 답글 인거같습니다!

>TL;DR Generally speaking, refs go against React's [declarative philosophy](https://stackoverflow.com/questions/33655534/difference-between-declarative-and-imperative-in-react-js), so you should use them as a last resort. Use state / props whenever possible.
[스택오브 플로우](https://stackoverflow.com/a/43345407)

Reply:
조금 이해했습니다! 당장 편할 수 있어도 react철학에 위배되고 당장은 편해도 나중 유지보수에 좋지 않다고 하니, 
한 번 ref 없이 만들어봐야겠네요!